src/pykeen/trackers/tensorboard.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+
+"""An adapter for TensorBoard."""
+import datetime
+import pathlib
+from typing import Any, Dict, Mapping, Optional
+
+from .base import ResultTracker
+from ..constants import PYKEEN_LOGS
+from ..utils import flatten_dictionary
+
+__all__ = [
+    'TensorBoardResultTracker',
+]
+
+
+class TensorBoardResultTracker(ResultTracker):
+    """A tracker for TensorBoard."""
+
+    def __init__(
+        self,
+        experiment_path: Optional[str] = None,
+        experiment_name: Optional[str] = None,
+        tags: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initialize result tracking via Tensorboard.
+
+        :param experiment_path:
+            The experiment path. A custom path at which the tensorboard logs will be saved.
+        :param experiment_name:
+            The name of the experiment, will be used as a sub directory name for the logging. If no default is given,
+            the current time is used. If set, experiment_path is set, this argument has no effect.
+        :param tags:
+            The additional run details which are presented as tags to be logged
+        """
+        from torch.utils.tensorboard import SummaryWriter as _SummaryWriter
+        self.SummaryWriter = _SummaryWriter
```

Comment:
Probably better to store with a more variable-like name

```suggestion
        self.summary_writer_cls = _SummaryWriter
```

Reply:
You can also add an instance variable annotation in the class body after the class docstring `self.summary_writer_cls: Type['torch.utils.tensorboard.SummaryWriter']` in combination with an import inside `typing.TYPE_CHECKING`. If you're not familiar with this, I will make a commit to add it for you - it'll be quite obvious once you see it but a bit tricky to come up with yourelf

Reply:
Thanks Charlie! Would be great if you could make the commit so I can see how it is done.

Reply:
See 63eb1a4